### PR TITLE
Use custom error classes instead of generic errors

### PR DESCRIPTION
**Pull request by joelhawksley**
### What are you trying to accomplish?

The goal of this PR is to add all of our error messages to our docs. In order to do so, I moved all of our errors to an `Errors` module and had YARD look for classes ending in `Error` that have a `MESSAGE` constant.

I also cleaned up a few errors that should have been removed previously. I'll follow up with a review of our error messages themselves.

What do y'all think of the names for these errors?

### CODE REVIEW

File: app/controllers/view_components_system_test_controller.rb [hunk 1]
```diff
@@ -24,7 +24,7 @@ def validate_file_path
     base_path = ::File.realpath(self.class.temp_dir)
     @path = ::File.realpath(params.permit(:file)[:file], base_path)
     unless @path.start_with?(base_path)
-      raise ArgumentError, "Invalid file path"
+      raise ViewComponent::SystemTestControllerNefariousPathError
```
Comment (camertron) [inline app/controllers/view_components_system_test_controller.rb:27]: Lol "nefarious"

File: lib/view_component/errors.rb [hunk 2]
```diff
@@ -1,16 +1,213 @@
...
+    def initialize
+      super(self.class::MESSAGE)
+    end
+  end
+
+  class DuplicateSlotContentError < StandardError
+    MESSAGE =
+      "It looks like a block was provided after calling `with_content` on COMPONENT, " \
+      "which means that ViewComponent doesn't know which content to use.\n\n" \
+      "To fix this issue, use either `with_content` or a block."
```
Comment (camertron) [inline lib/view_component/errors.rb:12]: Would these look better with squiggly heredocs?

```suggestion
    MESSAGE = <<~MESSAGE
      It looks like a block was provided after calling `with_content` on COMPONENT,
      which means that ViewComponent doesn't know which content to use.

      To fix this issue, use either `with_content` or a block.
    MESSAGE
```

**Review by camertron — APPROVED** [review]
Oh wow this is awesome! 🎉

File: lib/view_component/errors.rb (hunk 2 above)
Comment (joelhawksley) [reply lib/view_component/errors.rb:12]: I'm not sure these lines are equivalent, though? I'm trying to have the first sentence of the output be a single line of text.

Comment (joelhawksley) [reply lib/view_component/errors.rb:12]: Confirmed in the console: `=> "It looks like a block was provided after calling `with_content` on COMPONENT,\n`

**Pull request merged**